core/app/models/schedule.rb

```diff
@@ -18,10 +19,46 @@ def self.from_sections sections
     end
   end
 
+  def find_average_start
+      average_start = 0
+
+      first_periods = Hash.new(9999999999999)
+
+      num_days = 0
+      sections.each do |this_section|
+        this_section.periods.each do |p|
+          num_days += 1.0 if (last_periods[p["day"]] == -1)
+          first_periods[p["day"]] = p["start"].to_i if (p["start"].to_i < first_periods[p["day"]])
+        end
+      end
+      first_periods.each { |day, time| average_start += time }
+
+      average_start /= num_days
+  end
+
+  def find_average_finish
+    average_finish = 0
+
+    last_periods = Hash.new(-1)
+
+    num_days = 0
+    sections.each do |this_section|
+      this_section.periods.each do |p|
+        num_days += 1.0 if (last_periods[p["day"]] == -1)
+        last_periods[p["day"]] = p["end"].to_i if (p["end"].to_i > last_periods[p["day"]])
+      end
+    end
+    last_periods.each { |day, time| average_finish += time }
+
+    average_finish /= num_days
+  end
+
   def initialize params
     super
     @sections ||= []
     @uuid ||= SecureRandom.uuid
+    @average_start ||= 0
```

Comment:
set these when calling the relevant methods

Reply:
So I have the
`@average_start ||= 0`
`@average_finish ||= 0`
lines, but I also have

`average_start = 0`
`average_finish = 0`

in the beginning of their respective function. Is that okay?

Reply:
No, this will not work. `@some_variable` and `some_variable` are two completely different things. `@` means it's an instance variable, no `@` means it is a local